jerry-core/ecma/base/ecma-globals.h

```diff
@@ -799,17 +799,29 @@ typedef struct
 } ecma_object_t;
 
 /**
- * Description of built-in properties of an object.
+ * Description of built-in properties of an built-in ECMA-object.
  */
 typedef struct
 {
-  uint8_t id; /**< built-in id */
+  uint8_t id; /**< built-in object id */
   uint8_t length_and_bitset_size; /**< length for built-in functions and
                                    *   bit set size for all built-ins */
   uint16_t routine_id; /**< routine id for built-in functions */
   uint32_t instantiated_bitset[1]; /**< bit set for instantiated properties */
 } ecma_built_in_props_t;
 
+/**
+ * Description of built-in routine property of an built-in ECMA-object.
+ */
+typedef struct
+{
+  uint8_t built_in_object_id; /**< built-in object id */
+  uint8_t length; /**< length of built-in routine */
+  uint16_t id; /**< id of built-in routine */
+  uint16_t name; /**< name of built-in routine */
+  uint16_t bitset; /**< bit set for instantiated properties of builtin routine */
+} ecma_built_in_routine_props_t;
```

Comment:
I am not sure what was the discussion here, but `ecma_builtin_function_is_routine` silently expects that `id` is mapped to the same position as `routine_id`. I usually prefer unions, since the common fileds are ensured to be in the same position. What do you think?

Reply:
It's my mistake.
Fixed it as you commented. Thanks:)